distributed/core.py

```diff
@@ -346,16 +485,17 @@ def start_periodic_callbacks(self):
         """
         self._last_tick = time()
 
-        def start_pcs():
+        async def start_pcs():
             for pc in self.periodic_callbacks.values():
                 if not pc.is_running():
                     pc.start()
 
-        self.io_loop.add_callback(start_pcs)
```

Comment:
this was originally written so that `start_periodic_callbacks` can be run from any thread

Reply:
I wasn't aware of that. Is there a need to keep this as-is or can we safely simplify this? cc @fjetter?

Reply:
Yes that's correct, currently all our calls of start_periodic_callbacks are called from the running loop

That's why I think it's better to use an assertion that we're on the correct loop than support running from any thread

Reply:
Alright, I'll adjust according to the outline above then.